stream_alert/athena_partition_refresh/main.py

```diff
@@ -0,0 +1,241 @@
+        LOGGER.debug('Executing query: %s', kwargs['query'])
+        query_execution_resp = self.athena_client.start_query_execution(
+            QueryString=kwargs['query'],
+            QueryExecutionContext={'Database': kwargs.get('database', self.DATABASE_DEFAULT)},
+            ResultConfiguration={'OutputLocation': '{}/{}'.format(
+                self.athena_results_bucket,
+                self.athena_results_key
+            )}
+        )
+
+        query_execution_result = self.check_query_status(
+            query_execution_resp['QueryExecutionId'])
+
+        if query_execution_result != 'SUCCEEDED':
+            LOGGER.error(
+                'The query %s returned %s, exiting!',
+                kwargs['query'],
+                query_execution_result)
+            return False, {}
+
+        query_results_resp = self.athena_client.get_query_results(
+            QueryExecutionId=query_execution_resp['QueryExecutionId'],
+        )
+
+        # The idea here is to leave the processing logic to the calling functions.
+        # No data being returned isn't always an indication that something is wrong.
+        # When handling the query result data, iterate over each element in the Row,
+        # and parse the Data key.
+        # Reference: http://bit.ly/2tWOQ2N
+        if not query_results_resp['ResultSet']['Rows']:
+            LOGGER.debug('The query %s returned empty rows of data', kwargs['query'])
+
+        return True, query_results_resp
+
+    def check_database_exists(self, **kwargs):
+        """Verify the StreamAlert Athena database exists."""
+        database = kwargs.get('database', self.DATABASE_STREAMALERT)
+        query_success, query_resp = self.run_athena_query(
+            query='SHOW DATABASES LIKE \'{}\';'.format(database),
+        )
+
+        if query_success and query_resp['ResultSet']['Rows']:
+            return True
+        else:
+            LOGGER.error('The \'%s\' database does not exist. '
+                         'Create it with the following command: \n'
+                         '$ python stream_alert_cli.py athena create-db',
+                         database)
+            return False
+
+    def check_table_exists(self, table_name):
+        """Verify a given StreamAlert Athena table exists."""
+        query_success, query_resp = self.run_athena_query(
+            query='SHOW TABLES LIKE \'{}\';'.format(table_name),
+            database=self.DATABASE_STREAMALERT
+        )
+
+        if query_success and query_resp['ResultSet']['Rows']:
+            return True
+        else:
+            LOGGER.info('The streamalert table \'%s\' does not exist. '
+                        'For alert buckets, create it with the following command: \n'
+                        '$ python stream_alert_cli.py athena create-table '
+                        '--type alerts --bucket s3.bucket.id',
+                        table_name)
+            return False
+
+    def normal_partition_refresh(self):
```

Comment:
The only issue is that both are updating hive partitions, it's the way in which it happens that's different.  `MSCK REPAIR TABLE` vs `ADD PARTITION`